### PR TITLE
Cok chlorine edits

### PR DESCRIPTION
**Pull request by davidcok**
Grammatical and explanatory edits.

### CODE REVIEW

File: example-lt-modern.tex [hunk 1]
```diff
@@ -21,4 +21,5 @@
   towards \lstinline[style=c]|int| would also be acceptable.
 \item \lstinline[style=c]|\result != 0 <==> a < b| is acceptable because it is an
   equivalence between two predicates.
+ependencies
```
Comment (yakobowski) [inline example-lt-modern.tex]: Typo?

File: speclang_modern.tex [hunk 2]
```diff
@@ -661,6 +663,15 @@ \subsubsection{Exact computations}
 \end{listing-nonumber}
 \end{example}
 
+Note that the semantics of these constructs implies that in concept two different executions of the
```
Comment (vprevosto) [inline speclang_modern.tex]: This is a bit more complicated than that if the control flow might be influenced by the precision of the computation (i.e. if there are unstable tests where mathematical real arithmetic and floating point computation would give opposite answer). Indeed, execution is supposed to always follow the "normal" execution path as defined by C semantics (i.e. floating point computation): there is no such thing as a pure mathematical execution when it comes to branches.

File: speclang_modern.tex [hunk 3]
```diff
@@ -728,10 +739,9 @@ \subsection{Structures, Unions and Arrays in logic}
 \label{sec:aggregate}
 
 Aggregate C objects (i.e. structures, unions and arrays) are also
-possible values for terms in logic. They can be passed as parameters (and
-also returned) to logic functions, tested for equality, etc. like any
+possible values for terms in logic. They can be passed as parameters to and returned from logic functions, tested for equality, etc. like any
 other values.
-
+location
```
Comment (vprevosto) [inline speclang_modern.tex]: Typo?

File: speclang_modern.tex [hunk 4]
```diff
@@ -1278,7 +1287,7 @@ \subsection{Memory locations and sets of terms}
   value \lstinline|b| of binders
   satisfying predicate \lstinline|P|
   (binders \lstinline|b| are bound in both \lstinline|s| and \lstinline|P|).
-\item \lstinline|x \in s| holds if and only if is an element of s.
+\item \lstinline|x \in s| holds if and only if \lstinline|x| is an element of \lstinline|s| (that is, \lstinline|s$_1$| is a subset of \lstinline|s$_2$|);
```
Comment (vprevosto) [inline speclang_modern.tex]: it looks like the parenthesis should rather go into the next item

**Comment by davidcok** [comment]
If you like we can omit the text I added on \exact, for now; or it ca be embellished now or later. However, some discussion of the semantics here is needed, if any tool builder is to have a hope of implementing it correctly.

File: speclang_modern.tex [hunk 5]
```diff
@@ -661,6 +663,15 @@ \subsubsection{Exact computations}
 \end{listing-nonumber}
 \end{example}
 
+Note that the semantics of these constructs implies that in concept two different executions of the 
+program take place in parallel: one using the numeric precision as given in the program and one using mathematical real arithmetic. $\verb|\exact|(x)$ then gives the difference between the two for a specific expression and program state. If the argument to $\verb|\exact|(x)$ is just a constant or 
+an expression dependent on just some local computations, then the scope of the two computations is
+correspondingly constrained. But if the argument depends on program variables whose data dependencies extend back into the program, then the effect of a single $\verb|\exact|(x)$
+construct is considerable. This has obvious performance implications for runtime assertion checking
```
[the remark this answers is not in the document]
Comment (vprevosto) [reply speclang_modern.tex]: moved into a separate branch

File: speclang_modern.tex (hunk 2 above)
Comment (vprevosto) [reply speclang_modern.tex]: moved into a separate branch

**Comment by vprevosto** [comment]
Well, I didn't mean to have another branch for the discussion on the explanations about `\exact`, as I think that we can find a refined formulation pretty quickly, but now that the branch exists, let's move said discussion there.

**Pull request merged**